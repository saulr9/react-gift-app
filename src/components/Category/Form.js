import { useState } from 'react';
import PropTypes from 'prop-types';
import Button from '../UI/Button/Button';
import styles from './Form.module.css';

function Form({ setCategories }) {
  const [inputValue, setInputValue] = useState('');
  const handleChange = (e) => {
    const { target } = e;
    setInputValue(target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit} className={styles.form}>
      <input
        className={styles.input}
        type="text"
        id="input_search"
        placeholder="Search category"
        value={inputValue}
        onChange={handleChange}
      />
      <Button
        className={styles.button}
        buttonText="Buscar"
        handleSubmit={handleSubmit}
      />
    </form>
  );
}
Form.propTypes = {
  setCategories: PropTypes.func.isRequired
};
export default Form;
