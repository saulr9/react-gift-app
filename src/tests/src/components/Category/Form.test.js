import { shallow } from 'enzyme';
// eslint-disable-next-line import/no-unresolved
import Form from 'components/Category/Form';

describe('Testing Category form component', () => {
  const setCategories = () => {};
  const wrapper = shallow(<Form setCategories={setCategories} />);
  test('Should render Form', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Should change input value', () => {
    // eslint-disable-next-line prefer-const
    let input = wrapper.find('#input_search');
    const value = 'vegeta';
    input.simulate('change', { target: { value } });
    input = wrapper.find('#input_search').props().value;

    expect(input).toBe(value);
  });
});
