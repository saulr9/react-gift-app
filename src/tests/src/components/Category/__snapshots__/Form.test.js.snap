// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Category form component Should render Form 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input"
    id="input_search"
    onChange={[Function]}
    placeholder="Search category"
    type="text"
    value=""
  />
  <Button
    buttonText="Buscar"
    className="button"
    handleSubmit={[Function]}
  />
</form>
`;
